### DISCUSSION TITLE
New Feature Adding metrics to Skywalking using OTEL

### DISCUSSION THREAD

**Discussion by AjinThankachan** [apache/skywalking#12188] [Q&A]
Hi Team,

- OAP Version: 9.7
- Application : React

I am trying to push the metrics like DOM,DPT,etc. to Skywalking using the opentelemetry. I am able to fetch multiple metrics and structure and format in Opentelemetry format using the SDK and now how can i push it to Skywalking DB and show it in the Skywalking DB? Also i am able to fetch the session replay option as well. Is there anyway i can show in the Skywalking? 
**Code For Metrics**
`const { MeterProvider } = require('@opentelemetry/metrics');
const meterProvider = new MeterProvider();
// Get the meter instance
const meter = meterProvider.getMeter('RUM-Metrics');
// Create a value recorder for load time
const loadTimeRecorder = meter.createValueRecorder('load_time', {
description: 'Current page load time',
});
//////////
const customCounter = meter.createCounter('custom_counter', {
description: 'A custom counter',
unit: '1',
});
// Record a value for the custom counter
customCounter.add(1);
// console.log(customCounter)
// Define custom gauge metric
const customGauge = meter.createValueObserver('custom_gauge', {
description: 'A custom gauge',
unit: '1',
}, (observerResult) => {
observerResult.observe(42);
});
//////////
// console.log(customGauge)
// console.log(loadTimeRecorder)
console.log(meter)
process.on('SIGINT', async () => {
console.log('Shutting down...');
await meterProvider.shutdown();
console.log('Meter provider stopped.');
process.exit(0);
});
`

Thanks in advance. If this works well i will add it in the feature and add as a proposal.

**Comment by wu-sheng** [comment]
Copy my reply from https://skywalking.apache.org/docs/main/latest/en/setup/backend/opentelemetry-receiver/

I am not an expert of OTLP node client. If you want to export data to us, we need the OTLP gRPC exporter, https://skywalking.apache.org/docs/main/latest/en/setup/backend/opentelemetry-receiver/.

I am not sure that client supports gRPC/Protobuf format directly, if not, you could choose OTLP collector to transfer the HTTP 1.x format to gRPC format. If it is supported by that node agent, you could simply point to OAP gRPC port(11800 by default)
